### DISCUSSION TITLE
LFO and development suggestions

### DISCUSSION THREAD

**Discussion by smutax** [marcel-licence/esp32_fm_synth#7] [General]
Hi there. Congratulations on this brilliant project! In your video presentation of this fm synth it says that LFO is not implemented yet, but i was able to find some references to LFO in the code of this project. Can you please clarify if LFO implemented already and if not, whether you plan to do so? Also as a suggestion for future development i would go for a little polyphony increase and LP filter. Many thanks.

**Comment by marcel-licence** [comment]
Hi, thank you. The LFO part is prepared (I've added the variable for processing). Currently there is no processing and you have only the pitch modulation with also uses a LFO. The implementation is a good idea (it is also missing in other projects). 
The polyphony is limited by the processing power of the ESP32 (I will make this project flexible in future to run on other boards too). In addition to that I could use int instead of float which can also increase the polyphony.
If you mean a master filter the code can be copied from the basic synth for example. 
Thank you for the ideas. I think an update will follow soon =)

**Reply by smutax** [reply]
Hi Marcel. Glad to hear of an upcoming update and your intention to include other boards 👍. I guess good LFO would be the one implemented in your chiptune synth "Modulation lfo waveforms: (sine, saw, square, pulse, tri, prn)" 😉Also, the master filter would be nice if a "per channel" filter not possible. I guess the project can have some modularity where features can be enabled/disabled depending on the board used and usage scenario. Many thanks and Merry Christmas to you and your family.

**Comment by smutax** [comment]
Hi Martin.
Had a thought recently. Wouldn't it be nice to implement a "drum mode"? Let say chromatic  midi notes received on channel 10 would be redirected to other channels and the rest of the settings for (drum)sound design could be handled as is, through their channels. So it can be driven from sequencer drum track. Also the ability to set choke groups for drums to mute each other and the maximum polyphony for drums, like drum modules allow. All the best and keep up a good work.

**Reply by marcel-licence** [reply]
@smutax yes, a drum mode sounds nice. In addition to that you could also make some of the effects often used by the Sega Mega Drive / Sega Genesis. It would require some more memory to store for each note a different configuration. Mute other notes shouldn't be a problem.

**Reply by smutax** [reply]
@marcel-licence Thanks for considering above. Happy Christmas.
